Cython/Shadow.pyi

```diff
@@ -100,3 +101,34 @@ class typedef(CythonType, Generic[_T]):
 #    def __init__(self) -> None: ...
 
 #def fused_type(*args: Tuple[_T]) -> Type[FusedType[_T]]: ...
+
+def typeof(arg: Any) -> str: ...
+
+_C = TypeVar('_C', bound='Callable')
+
+class _EmptyDecoratorAndManager(object):
+    def __call__(self, x: _C) -> _C: ...
+
+    def __enter__(self) -> None: ...
+
+    def __exit__(
+        self,
+        exc_type: Optional[type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType]
+    ) -> None: ...
+
+cclass = ccall = cfunc = inline = _EmptyDecoratorAndManager()
+
+@overload
```

Comment:
When we overloading a function we preserve one overload as it and decorate others with `typing.overload`.
You can look [here](https://mypy.readthedocs.io/en/stable/error_code_list.html?highlight=Overload#check-calls-to-overloaded-functions-call-overload)

Reply:
Your current code uses decorator for every overload

Reply:
I tried it, like this:
```py
@overload
def declare(
    t: Optional[Callable[..., _T]] = None,
    value: Any = ...,
    *,
    visibility: str = ...
) -> _T:
    ...

def declare(**kwargs: type) -> None:
    ...
```
And my static type checker of choice, Pyright, had this to say:
```
"declare" is marked as overload, but additional overloads are missing
```

Reply:
> Remember, we're in a stub file, it behaves slightly differently.

Yeah, that's probably the deal, sorry